### PR TITLE
Milestone 6: Friendships v2

### PR DESCRIPTION
**Pull request by SofiBretz**
### Building Facebook Milestone 6: Friendships v2

In this milestone we fixed the scenarios:

- When a user sends a request, the option to send another request should disappear.
- When a request is accepted, it should disappear from the notification bar.
- The friendship notifications are more visible.
- Rubocop fixed, no offenses detected.

### CODE REVIEW

File: app/controllers/friendships_controller.rb [hunk 1]
```diff
@@ -3,14 +3,14 @@
 class FriendshipsController < ApplicationController
   def create
     user = User.find(params[:user_id])
-    return current_user.send_friend_request(user) if user
+    current_user.send_friend_request(user) if user
```
Comment (maelfosso) [inline app/controllers/friendships_controller.rb]: You must add an additional row.
Read the indication from [milestone6](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/), please

**Comment by SofiBretz** [comment]
### Building Facebook Milestone 6: Friendships v2

In this milestone we:

- We fixed the list of the post for only the friends.
- We created 3 tables, one for receiver, sender, and indices to friendship.
- We created a migration to change the default value on the friendship table.
- Rubocop fixed, no offenses detected.

File: app/models/user.rb [hunk 2]
```diff
@@ -36,41 +36,40 @@ def unlike(post)
...
-    friends_array.compact
+    friendships = sent_requests.includes(:receiver).where(confirmed: true).references(:users)
+    friendships.map(&:receiver)
   end
 
   def pending_friends
-    sent_requests.map { |u| u.receiver unless u.confirmed }.compact
+    requests = sent_requests.includes(:receiver).where(confirmed: false).references(:users)
+    requests.map(&:receiver)
   end
```
Comment (samgaco) [inline app/models/user.rb]: :+1:

File: app/models/user.rb [hunk 3]
```diff
@@ -36,41 +36,40 @@ def unlike(post)
...
 
   def friend_requests
-    received_requests.map { |u| u.sender unless u.confirmed }.compact
+    requests = received_requests.includes(:sender).where(confirmed: false).references(:users)
+    requests.map(&:sender)
   end
```
Comment (samgaco) [inline app/models/user.rb]: :+1:

**Pull request merged**